common/attest_plugin.c

```diff
@@ -338,40 +347,106 @@ oe_result_t oe_verify_attestation_certificate_with_evidence(
 
     //------------------------------------------------------------------------
     // Validate the report's trustworthiness
+    // Verify the remote report to ensure its authenticity.
+    // set enclave to NULL because we are dealing only with remote report now
     //------------------------------------------------------------------------
 
     // determine the size of the extension
     if (oe_cert_find_extension(
```

Comment:
The code in the four if blocks are almost identical. I'd suggest making an array of OIDs and perform in a loop:

`
oid_array = { oid_report_1, oid_report_2, oid_evidence_1, oid_evidence_2 };
for (n = 0; n < countof(oid_array); n++) {
    report_size = 0;
    if (oie_cert_find_extension(..., oid_array[n], ..., &report_size) == TOO_SMALL) {
        ....
        break;
    } 
}

if (n <2)
    // is report
else if (n < 4)
    // is evidence
else
    // no match
`